src/unix/pipe.c

```diff
@@ -287,6 +287,62 @@ int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
 }
 
 
+int uv_pipe_get_credentials(const uv_pipe_t* handle, uv_credentials_t* creds) {
+  int r;
+  uv_os_fd_t os_fd;
+
+  r = uv_fileno((uv_handle_t*) handle, &os_fd);
+  if (r < 0)
+    return r;
+
+#if defined(__APPLE__)
+  {
+    /* https://stackoverflow.com/a/38207320 */
+    unsigned int uid;
```

Comment:
why not `uid_t` and `gid_t`?

Reply:
Oops didn't mean to resolve, sorry.

Because uid_t and gid_t are unsigned, and on Windows I wanted to return -1 to match the behavior of the getpasswd function.

Reply:
FWIW they were originally `uid_t` and `gid_t`.

Reply:
yes, but `unsigned int` is obviously also unsigned, so I don't see how that's better. It is, imho, preferable to match the type defined by `getpeereid()`, then do whatever conversion after - which you are doing anyways...

Reply:
@ptlomholt This is a local variable, not a return value. I agree with your reasoning but that prevents me from returning `-1`. The local variable _is_ `unsigned int` but that's required by Apple's API.